operations/mimir-mixin/dashboards/compactor.libsonnet

```diff
@@ -1,6 +1,128 @@
 local utils = import 'mixin-utils/utils.libsonnet';
 
+// Panel query override functions
+local overrideFieldByName(fieldName, overrideProperties) = {
+  matcher: {
+    id: 'byName',
+    options: fieldName,
+  },
+  properties: overrideProperties,
+};
+
+local overrideProperty(id, value) = { id: id, value: value };
+
+// Panel query value mapping functions
+local mappingRange(from, to, result) = {
+  type: 'range',
+  options: {
+    from: from,
+    to: to,
+    result: result,
+  },
+};
+
+local mappingSpecial(match, result) = {
+  type: 'special',
+  options: {
+    match: match,
+    result: result,
+  },
+};
+
+// Panel query transformation functions
+
+local transformation(id, options={}) = { id: id, options: options };
+
+local transformationCalculateField(alias, left, operator, right, replaceFields=false) =
+  transformation('calculateField', {
+    alias: alias,
+    binary: {
+      left: left,
+      operator: operator,
+      right: right,
+    },
+    mode: 'binary',
+    replaceFields: replaceFields,
+  });
+
+
+// This applies to the "longest time since successful run queries"
+local fixTargetsForTransformations(panel, refIds) = panel {
+  // Make some adjustments to the targets to make them compatible with the transformations required
+  targets: [
+    panel.targets[i] {
+      refId: refIds[i],
+      format: 'table',
+      instant: true,
+    }
+    for i in std.range(0, std.length(refIds) - 1)
+  ],
+};
+
 (import 'dashboard-utils.libsonnet') {
+
+  local vars = {
+    instance: $._config.per_instance_label,
+  },
+
+  local lastRunThresholds = {
+
+    local hours = 60 * 60,
+
+    // In terms of hours
+    delayed: 2 * hours,
+    late: 6 * hours,
+    veryLate: 12 * hours,
+
+    local _ = self,
+
+    // steps for thresholds
+    steps: [
+      { value: 0, color: 'green' },
+      { value: _.delayed, color: 'yellow' },
+      { value: _.late, color: 'orange' },
+      { value: _.veryLate, color: 'red' },
+    ],
+
+    // status mappings: messages and colors
+    mappings: [
+      mappingRange('-Infinity', 0, { color: 'transparent', text: 'N/A' }),
+      mappingRange(0, _.delayed, { color: 'green', text: 'Ok' }),
+      mappingRange(_.delayed, _.late, { color: 'yellow', text: 'Delayed' }),
+      mappingRange(_.late, _.veryLate, { color: 'orange', text: 'Late' }),
+      mappingRange(_.veryLate, 'Infinity', { color: 'red', text: 'Very late' }),
+      mappingSpecial('null+nan', { color: 'transparent', text: 'Unknown' }),
+    ],
+  },
+
+  local lastRunQuery =
```

Comment:
You need to filter metrics in the query using `$.jobMatcher($._config.job_names.compactor)` (see other queries in this dashboard as example).